src/pages/SignIn.tsx

```diff
@@ -0,0 +1,61 @@
+import { useForm, SubmitHandler } from 'react-hook-form'
+import './../css/form.css'
+import { Link } from 'react-router-dom'
+
+type Inputs = {
+  name: string
+  email: string
+  password: string
+  emailRequired: string
+}
+
+type SignUpProps = {
+  signIn: (input: User) => void
+}
+
+function SignIn({ signIn }: SignUpProps) {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<Inputs>()
+  const onSubmit: SubmitHandler<Inputs> = (data) => signIn(data)
+
+  watch('email')
+
+  return (
+    <div className="form-container">
+      <h2>Sign in</h2>
+      <form className="form" onSubmit={handleSubmit(onSubmit)}>
```

Comment:
A lot of times in Typescript it's easier to keep functions inline so types can be inferred & you don't need to duplicate a function type to later pass it in
```suggestion
      <form className="form" onSubmit={handleSubmit((data) => signIn(data))}>
```